### PR TITLE
benchmark ssdlite detection pipeline

### PR DESCRIPTION
**Pull request by pmeier**

### CODE REVIEW

File: results/20230403073901.log [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+           v2 / v1
+Tensor        1.66
+PIL           1.54
+
+               x / PIL, v1
+Tensor, v1            0.92
+Tensor, v2            1.53
+PIL, v1               1.00
+PIL, v2               1.54
+Datapoint, v2         1.52
```
Comment (pmeier) [inline results/20230403073901.log:291]: highlighting

File: main.py [hunk 2]
```diff
@@ -23,6 +23,10 @@ def write(self, message):
         self.stdout.write(message)
         self.file.write(message)
 
+    def flush(self):
```
Comment (NicolasHug) [inline main.py:26]: where is this needed?

**Review by NicolasHug — APPROVED** [review]

File: transforms.py [hunk 3]
```diff
@@ -135,3 +195,238 @@ def forward(self, img):
...
         return F_v2.crop(inpt, **params)
+
+
+class WrapCocoDetectionReferenceSampleForTransformsV2:
+    def __init__(self):
+        num_samples = 117_266
+        wrapper_factory = WRAPPER_FACTORIES[datasets.CocoDetection]
+        mock_dataset = SimpleNamespace(ids=list(range(num_samples)))
+        wrapper = wrapper_factory(mock_dataset)
+        self.wrapper = functools.partial(wrapper, num_samples // 2)
```
[the remark this answers is not in the document]
Comment (pmeier) [reply transforms.py:206]: This whole thing is mocking the dataset so we can re-use the dataset wrappers as transform. `num_samples = 117_266` is the number of samples in the filtered COCO dataset so each sample has one annotation. 

In the wrapper, we get passed the `idx` of the sample

https://github.com/pytorch/vision/blob/39712558247e0d806ceb0df384a6258aa744f950/torchvision/datapoints/_dataset_wrapper.py#L149

and use it like this:

https://github.com/pytorch/vision/blob/39712558247e0d806ceb0df384a6258aa744f950/torchvision/datapoints/_dataset_wrapper.py#L284

Since we don't have access to the `idx` here, I chose a proxy. Since `ids` is a list, I used the half of the maximum, to have on average the same performance as doing the correct lookup.

File: transforms.py [hunk 4]
```diff
@@ -135,3 +195,238 @@ def forward(self, img):
...
+        # for conversion to coco api
+        area = torch.tensor([obj["area"] for obj in anno])
+        iscrowd = torch.tensor([obj["iscrowd"] for obj in anno])
+        target["area"] = area
+        target["iscrowd"] = iscrowd
+
+        return image, target
+
+
+class DetectionReferenceRandomHorizontalFlipV1(transforms_v1.RandomHorizontalFlip):
```
Comment (NicolasHug) [inline transforms.py:299]: In order to ease per-transform comparison visually, perhaps keep the shorter name `RandomHorizontalFlipV1` (same for other transforms).

Ideally we would even drop the V1/V2 suffix

**Comment by pmeier** [comment]
We identified the slowdown from detection v2 stems from two things:

1. `pytree` is comparatively really slow for "large" inputs. Since by default, `CocoDetection` includes a nested list of coordinates. Although they are not transformed, they also need to be flattened and unflattened.
2. Keeping the masks in the sample means they will be transformed although we don't need them.

Results here are run on torchvision nightly April 4. When pytorch/vision#7488 is merged and pytorch/vision#7489 addressed, we need to re-run.
